raft/rawnode.go

```diff
@@ -172,9 +175,6 @@ func (rn *RawNode) HasReady() bool {
 // Advance notifies the RawNode that the application has applied and saved progress in the
 // last Ready results.
 func (rn *RawNode) Advance(rd Ready) {
-	if !IsEmptyHardState(rd.HardState) {
```

Comment:
@mitake what are these changes related to? Thanks!

Reply:
@mitake I see a second commit signed-by Nathan. I think we should make such changes separately on a clean PR? Thanks!

Reply:
+1

@mitake you need to rebase this PR.

Reply:
@spzala @ahrtr oh sorry, I rebased on the latest main

Reply:
Thanks @mitake @ahrtr